themes/default/content/blog/2022-03-22-my-pulumi-managing-my-dns/index.md

```diff
@@ -0,0 +1,249 @@
+---
+title: "My Pulumi: Managing My DNS"
+date: 2022-03-22T10:00:00Z
+draft: false
+meta_desc: In this aritcle, Rawkode shows how he using Pulumi to manage his DNS records for his many domains
+meta_image: meta.png
+authors: ["david-flanagan"]
+tags: ["Domains", "DNS"]
+---
+
+Hello, my name is David Flanagan and I own more domains than I need. The problem is, I have too many ideas; and as we all know, ideas don't become real until you buy the domain name. Unfortunately, more often than not, that's about as far as my ideas go - because, **life**. That being said, I do try to keep my DNS records under control in the event that life affords me the time to follow-up on one of these ideas. Today, I want to show you how I do that.
+
+## Overview
+
+Using Pulumi's ComponentResource API, we're able to provide a nice developer experience for manipulating DNS records through methods on an object - we call this object the "Domain Controller". I have a domain controller for every domain that I own. Through this controller I can add DNS records through helper methods that keep the interface as simple as can be. Let's take a look.
+
+### Fetching the Controller
+
+First, we need to request a controller for our domain name we want to manipulate. This creates a new object based on our `Controller` class.
+
+```typescript
+const rawkodeDev = new Controller("rawkode.dev");
+```
+
+### Creating Records
+
+Now that we have a controller for the `rawkode.dev` domain, we can start to add records. As with pretty much every domain, there's a selection of things required out of the box:
+
+- MX records for email configuration
+  - or an SPF record to note that the domain will not be used for email
+- TXT records for dmarc and other verification purposes
+- A/CNAME records for any subdomains in-use
+
+#### Configuring Apex
+
+My website is delivered via [Vercel](https://vercel.com), so I need an A record configured with their BGP-advertised ingress address. While I prefer to host my website as [no-www](https://dropwww.com/), I do create a `www` record to redirect as required.
+
+```typescript
+zone.createRecord("@", "A", ["76.76.21.21"]);
```

Comment:
It might be nice to link to the repo before introducing these snippets (I see it's linked later), as without that, it's unclear what's being referred to.

Reply:
Ah, I see now that the walkthrough comes later. In that case, it might be nice to briefly mention somewhere above that "we'll look at the code in more detail later", or similar -- again just so the reader isn't left wondering what the snippets refer to.